pkg/bloombuild/builder/builder.go

```diff
@@ -477,9 +483,10 @@ func (b *Builder) loadWorkForGap(
 	return seriesItr, blocksIter, nil
 }
 
-// TODO(owen-d): pool, evaluate if memory-only is the best choice
-func (b *Builder) rwFn() (v1.BlockWriter, v1.BlockReader) {
-	indexBuf := bytes.NewBuffer(nil)
-	bloomsBuf := bytes.NewBuffer(nil)
-	return v1.NewMemoryBlockWriter(indexBuf, bloomsBuf), v1.NewByteReader(indexBuf, bloomsBuf)
+func (b *Builder) writerReaderFunc() (v1.BlockWriter, v1.BlockReader) {
+	dir, err := os.MkdirTemp("", "bloom-block-")
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Is it necessary to make the working directory configurable? The temp directory seems to be the appropriate place to write blocks to before they are uploaded.

Reply:
I think the tmp dir approach should be fine to start with but having a configurable path would allow us to use SSDs for example to speed IO up if it becomes a problem. So I'd have a hidden config that defaults to empty in which case, we use a tempdir.